### PR TITLE
Remove track mod

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Based on #2107

Interesting that i had to modify the economy function. I'm surprised we haven't passed a signed cost factor yet into that function (or maybe we have and never noticed the cost is calculated wrong!)

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
> Interesting that i had to modify the economy function. I'm surprised we haven't passed a signed cost factor yet into that function (or maybe we have and never noticed the cost is calculated wrong!)

That reminds me that, in very early versions of OpenLoco, removing signals would yield billions of currency to the player. That was obviously solved, but sounds like it could be related somehow.

**Comment by duncanspumpkin** [comment]
> > Interesting that i had to modify the economy function. I'm surprised we haven't passed a signed cost factor yet into that function (or maybe we have and never noticed the cost is calculated wrong!)
> 
> That reminds me that, in very early versions of OpenLoco, removing signals would yield billions of currency to the player. That was obviously solved, but sounds like it could be related somehow.

Yeah there is a hook for that that sets one register to a high value. I can't remember why it solves the issue anymore though

**Review by AaronVanGeffen — APPROVED** [review]
Appears to be working properly. It's a pity you can only remove track additions one tile at a time, though. I hope that's something we can expand on in the future.

**Comment by duncanspumpkin** [comment]
> Appears to be working properly. It's a pity you can only remove track additions one tile at a time, though. I hope that's something we can expand on in the future.

Track mod ghosts are removed multi track block. That's the main way i tested it.

**Comment by AaronVanGeffen** [comment]
> Track mod ghosts are removed multi track block. That's the main way i tested it.

Oh man, I wish we could expose that to players, too. Anyway, ghosts appear to be working properly for block sections and full layout, too.

**Comment by duncanspumpkin** [comment]
> > Track mod ghosts are removed multi track block. That's the main way i tested it.
> 
> Oh man, I wish we could expose that to players, too. Anyway, ghosts appear to be working properly for block sections and full layout, too.

We could just make it that if the track mod window is open and the multi track is selected then remove multi track.

**Comment by AaronVanGeffen** [comment]
> We could just make it that if the track mod window is open and the multi track is selected then remove multi track.

If that is trivial to do, I think that would be a really nice feature!

**Comment by duncanspumpkin** [comment]
> Based on #2107
> 
> Interesting that i had to modify the economy function. I'm surprised we haven't passed a signed cost factor yet into that function (or maybe we have and never noticed the cost is calculated wrong!)

So it turned out that that economy mistake is the source of the mysterious ridiculous cost for signals when upgrading them! So I've added a changelog entry for that.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 23.09+ (???)
 ------------------------------------------------------------------------
+- Fix: [#2110] Very large signal placement cost when upgrading a signal.
```
Comment (AaronVanGeffen) [inline CHANGELOG.md:3]: Could you swap these lines around so they're in linear order again?

**Pull request merged**